analysis/quantifier-comparisons/12-spatial-transcriptomics.Rmd

```diff
@@ -0,0 +1,595 @@
+
+## Set Up
+
+```{r}
+library(magrittr)
+library(ggplot2)
+library(SingleCellExperiment)
+library(SpatialExperiment)
+library(ggspavis)
+library(ggupset)
+```
+
+```{r}
+# load in benchmarking functions that will be used for copying data and generating sample tables
+function_path <- file.path(".." ,"benchmarking-functions", "R")
+file.path(function_path, list.files(function_path, pattern = "*.R$")) %>%
+  purrr::walk(source)
+```
+
+
+```{r}
+# set up file paths 
+base_dir <- here::here()
+
+# output folder to store alevin-fry and cellranger quants from S3
+data_dir <- file.path(base_dir, "data", "spatial") 
+s3_out <- file.path(data_dir, "data", "quants")
+
+# output folder for image files 
+image_dir <- file.path(data_dir, "data", "images")
+
+# results directory 
+results_dir <- file.path(data_dir, "results")
+
+# create directories
+if(!dir.exists(data_dir)){
+  dir.create(data_dir, recursive = TRUE)
+}
+
+if(!dir.exists(image_dir)){
+  dir.create(image_dir, recursive = TRUE)
+}
+
+if(!dir.exists(results_dir)){
+  dir.create(results_dir, recursive = TRUE)
+}
+```
+
+
+```{r}
+mito_file <- file.path(base_dir, "sample-info", "Homo_sapiens.GRCh38.103.mitogenes.txt")
+  
+# read in mito genes 
+mito_genes <- readr::read_tsv(mito_file, col_names = "gene_id")
+mito_genes <- mito_genes %>%
+  dplyr::pull(gene_id) %>%
+  unique()
+```
+
+
+## Copy data from S3
+
+```{r}
+# grab alevin fry and cellranger output
+sample_ids <- c("SCPCR000372", "SCPCR000373")
+
+aws_copy_samples(local_dir = s3_out,
+                 s3_dir = "s3://nextflow-ccdl-results/scpca",
+                 samples = sample_ids,
+                 tools = c("alevin-fry-knee", "cellranger"))
+```
+
+## Comparing SpatialExperiment to Seurat
+
+### Importing Alevin-fry and Spaceranger output into a SpatialExperiment
+
+First we are testing use of a `SpatialExperiment` object to store the spatial transcriptomics results. 
+We will create two functions, one to deal with creating the combined object from both Alevin-fry and Spaceranger output, and the second to import Spaceranger output only. 
+
+```{r}
+# custom function for converting Alevin-fry and Spaceranger output to spe
+convert_fry_spaceranger_spe <- function(fry_dir, spaceranger_dir, sample_name){
+  fry_sce <- scpcaTools::read_alevin(fry_dir, 
+                                     usa_mode = TRUE,
+                                     which_counts = "spliced")
+  
+  # read in image data
+  image_data <- SpatialExperiment::readImgData(spaceranger_dir,
+                            sample_id = sample_name)
+  
+  # read in spatial coordinates 
+  spatial_data_file <- file.path(spaceranger_dir, "tissue_positions_list.csv")
+  spatial_data <- readr::read_csv(spatial_data_file, col_names = c("barcode", "in_tissue", "array_row",
+                                                                   "array_col","pxl_row_in_fullres",
+                                                                   "pxl_col_in_fullres"))
```

Comment:
Can we skip included the `col_names` here? Part of this is just a general preference, as this always seems brittle, but I am also trying to figure out why some of your images have flipped axes relative to one another. I doubt that this is the reason, but it is a distraction for comparisons.

Reply:
I think we want to add the column names here because this file doesn't have any column names and the later functions rely on the spatial data having specific names so we have to add them in somewhere. I do think the flip might be the order of the coordinates later on.

Reply:
That makes sense. I might pull them out to a separate variable, just for clarity/formatting.